### PR TITLE
Warn about third-party JVM agents in docs

### PR DESCRIPTION
**Pull request by williamrandolph**
We've had reports of cases where Elasticsearch nodes froze mysteriously or had difficult-to-diagnose performance problems, and the eventual culprit was some monitoring agent that attached into to the JVM and interfered with what Elasticsearch needed to do. We can add a warning about this to our docs so that we have something to reference when this problem comes up again.

### CODE REVIEW

File: docs/reference/setup/install.asciidoc
Comment (rjernst) [inline docs/reference/setup/install.asciidoc:90]: Remove "long-running"? Any java agent, regardless of how long lived, should not be attached to Elasticsearch.

**Review by rjernst — COMMENTED** [review]
Maybe this paragraph should be in its own section? I think it may get lost underneath JVM, and it isn't really about selection of the jvm as the other documentation here is about.

File: docs/reference/setup/install.asciidoc
Comment (DaveCTurner) [inline docs/reference/setup/install.asciidoc:92]: I think we should have fewer qualifiers here. It's not just _long-running agents for instrumentation_ that we want to exclude, running with any kind of agent attached is unsupported.

Also I'd prefer stronger wording than "you should not" - ideally "do not" or something equally imperative.

**Review by rjernst — APPROVED** [review]
LGTM

**Review by DaveCTurner — APPROVED** [review]
LGTM2

**Comment by williamrandolph** [comment]
Thanks! I'll merge once @jrodewig can approve it.

File: docs/reference/setup/install.asciidoc
Comment (jrodewig) [inline docs/reference/setup/install.asciidoc:95]:
```suggestion
=== JVM and Java agents
```

File: docs/reference/setup/install.asciidoc
Comment (jrodewig) [inline docs/reference/setup/install.asciidoc:97]:
```suggestion
Don't use third-party Java agents that attach to the JVM. These agents
```

File: docs/reference/setup/install.asciidoc
Comment (jrodewig) [inline docs/reference/setup/install.asciidoc:98]:
```suggestion
can reduce {es} performance, including freezing or crashing nodes.
```

**Review by jrodewig — APPROVED** [review]
Approved. I left some suggestions. Please accept the fix to the heading case. You can take or leave the rest as you see fit. Thanks!

**Pull request merged**
